File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F12.P2.kt
fun transform(o: Int?): Int {
    // BEGIN PROBE F12.P2
    val h = { v: Int -> v + 1 }
    val p: Int? = o?.let(h)
    return p ?: 0
    // END PROBE F12.P2
}

fun main() {
    println(transform(null))
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/python/F12.P2.py
def transform() -> int:
    o: int | None = None
    # BEGIN PROBE F12.P2
    def h(v):
        return v + 1

    p: int | None = None if o is None else h(o)
    return 0 if p is None else p
    # END PROBE F12.P2


print(transform())
